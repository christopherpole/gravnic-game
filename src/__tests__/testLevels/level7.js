import {
  MOVE_RIGHT,
  MOVE_DOWN,
  MOVE_NONE,
  ENTITIES,
  levelIsComplete,
  makeMoves,
} from '../../index';

describe('Level 7', () => {
  it('Matches the correct snapshot', () => {
    const gameState = [
      [
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 2, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#ff0000', id: 1 },
        },
        {
          staticEntity: { id: 4, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#008000', id: 3 },
        },
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
      ],
      [
        { staticEntity: null, movableEntity: null },
        { staticEntity: { id: 5, entityId: ENTITIES.FLOOR }, movableEntity: null },
        { staticEntity: { id: 6, entityId: ENTITIES.FLOOR }, movableEntity: null },
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
      ],
      [
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 8, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.RAINBOW_BLOCK, id: 7 },
        },
        {
          staticEntity: { id: 10, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.GLASS, id: 9 },
        },
        { staticEntity: { id: 11, entityId: ENTITIES.FLOOR }, movableEntity: null },
        { staticEntity: { id: 12, entityId: ENTITIES.FLOOR }, movableEntity: null },
      ],
      [
        {
          staticEntity: { id: 14, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#800080', id: 13 },
        },
        { staticEntity: { id: 15, entityId: ENTITIES.FLOOR }, movableEntity: null },
        { staticEntity: { id: 16, entityId: ENTITIES.FLOOR }, movableEntity: null },
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 18, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#008000', id: 17 },
        },
      ],
    ];

    const gameStates = makeMoves(gameState, [MOVE_NONE, MOVE_DOWN, MOVE_RIGHT]);
    expect(gameStates).toMatchSnapshot();
    expect(
      levelIsComplete(
        gameStates[gameStates.length - 1][gameStates[gameStates.length - 1].length - 1],
      ),
    ).toBe(true);
  });
});
