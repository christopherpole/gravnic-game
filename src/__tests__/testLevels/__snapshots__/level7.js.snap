// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level 7 Matches the correct snapshot 1`] = `
Array [
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 13,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 13,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 13,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "fading": true,
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "fading": true,
            "id": 13,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "RAINBOW_BLOCK",
            "fading": true,
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "fading": true,
            "id": 3,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "fading": true,
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "GLASS",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
    ],
  ],
]
`;
